Panic bunker + new player notification configs

MrsTonedOne opened the pull request:
Panic bunker! Uses the power of the database to reject new ckeys that have never connected to the server(s) before.
Admins are exempt, even while deadmin'ed, but they are also unlikely to be new ckeys except in rare cases that involve vv fuckary.

Its both a round only +server verb, and a config option for more long term panic attacks.

New player notification is now configurable. You can change how long they happen for.

Ie: First connection only, first x number of days on the server or never

Adds an optional default disabled irc notification of a players first time connecting.
This will only fire if first connection admin alerts above are turned on AND there are no valid admins on and not afk.

MrsTonedOne committed:
Panic bunker + new player notification configs

Panic bunker! Uses the power of the database to reject new ckeys that have never connected to the server(s) before.
Admins are exempt, even while deadmin'ed, but they are also unlikely to be new ckeys except in rare cases that involve vv fuckary.

New player notification is now configurable.

Adds a optional default disabled irc notification of a players first time connecting.

MrStonedOne commented on code/modules/admin/verbs/panicbunker.dm:
```diff
@@ -0,0 +1,16 @@
+		usr << "<span class='adminnotice'>The Database is not enabled!</span>"
+		return
+
+	if (!dbcon || !dbcon.IsConnected())
```
todo: make it so you can turn it off while the database is not connected, or flat out remove some of these checks

paprka commented:
it should redirect them to a hub server too come on you need to go full /vg/

MrStonedOne commented:
@paprka if you are just gonna shit post on my prs just fucking go.

What is the point? stalk me around github? Is that suppose to annoy me?

Because its not going to work.

All i'm going to do is go to github and report you for harassing me.

You've already been kicked out from this community, the only reason you can still comment here is because github a shit and requires you to put up a support ticket to issue bans to repos attached to organisations rather than personal accounts.

Dorsisdwarf commented:
>paprka hounded out of the community by the lynch mob

>Still makes useful ease-of-life PRs for the community

>"Halp he is stalk me"

dannno commented:
is this really what you think harassment is
where do you think we are

MrsTonedOne committed:
Makes the irc new player alert option behave like irc admin pm notices
It will only trigger if the new player alert is enabled for admin display and no valid non-afk admins are on to see it.

MrsTonedOne committed:
Adds panic bunker changelog

MrsTonedOne committed:
Tweaks panic bunker verb logic a little bit
Allows toggling even if the database is not connected, alerts admins instead.

MrsTonedOne committed:
Fixes guests being allowed to connect while panic bunker is active
I was tempted to move all the panic bunker shit to isbanned(), but doing so would mean adding yet ANOTHER query that gets run on every connection, and run for every player on world restart.

That just seemed like not worth the effort to just make panic bunker trigger a little bit earier.

MrsTonedOne committed:
Adds failed login message for panic bunker.

MrStonedOne commented:
Alright, I did a bunch more testing while on more sleep, should be ready once this hits 24 hours.

Razharas commented:
What for such a thing needed?

MrStonedOne commented:
Admins have requested it. Every now and then we get a really determined ban evader, who figured out how to spoof cids or some shit. so they just keep connecting and shit.

this would allow admins to temporarily disable new users from connecting to the server while not preventing their regulars from connecting.

Also helps for when the game gets links on major sites to prevent the wave of griffers
